sources/engine/Stride.UI/Rendering/UI/UIRenderFeature.cs

```diff
@@ -54,29 +47,23 @@ protected override void InitializeCore()
 
             Name = "UIComponentRenderer";
             game = RenderSystem.Services.GetService<IGame>();
-            input = RenderSystem.Services.GetService<InputManager>();
             uiSystem = RenderSystem.Services.GetService<UISystem>();
-            graphicsDeviceService = RenderSystem.Services.GetSafeServiceAs<IGraphicsDeviceService>();
 
+            // TODO: Change Batch to be local to the UIRenderFeature and remove dep on UISystem. Requires refactoring ElementRenderer.
             if (uiSystem == null)
             {
-                var gameSytems = RenderSystem.Services.GetSafeServiceAs<IGameSystemCollection>();
+                var gameSystems = RenderSystem.Services.GetSafeServiceAs<IGameSystemCollection>();
+                
                 uiSystem = new UISystem(RenderSystem.Services);
                 RenderSystem.Services.AddService(uiSystem);
-                gameSytems.Add(uiSystem);
+                gameSystems.Add(uiSystem);
             }
 
             rendererManager = new RendererManager(new DefaultRenderersFactory(RenderSystem.Services));
 
             batch = uiSystem.Batch;
         }
 
-        partial void PickingPrepare();
-
-        partial void PickingUpdate(RenderUIElement renderUIElement, Viewport viewport, ref Matrix worldViewProj, GameTime drawTime, ref UIElement elementUnderMouseCursor);
-
-        partial void PickingClear();
-
```

Comment:
I assume you mean the thought process for moving the picking from the draw method to the update method?
There are a couple of reason, and my general thought process. 

1. All the input being handled in a single place allows for a single source of truth. Which in turn, makes it easier to find and edit the all input code (for example, if you wanted to block/disable all UI input for whatever reason). Have an easy to debug and consistent execution flow of input events.
2. Allows for the `UIRenderFeature` to be responsible for only rendering instead of gameplay funcinality. Making it consistent with other `RenderFeatures` and arguably what a user might expect when adding it.
3. Allows for the UI to accept input and be fully functional even if it is not rendered, meaning they can be more 'lookless'. Which should make running tests and debugging easier, with less points for failure.
4. When it is in the `Draw` loop, all methods that the event raises will also be execute in the `Draw` loop. This includes, `PointerPressed`, `PouinterReleased`, `PointerOver`, `PointerMove`, `PointerEnter`, and `PointerLeave`. And those all can, and do mutate the UIElements, including adding or removing new UIElements.
5. _If_ the `Draw` and `Update` loops are not synced (always happening one right after the other), then pointer inputs could either be missed or processed multiple times. I tried to track when and where `Draw` is called, but couldn't get a clear answer. So I am not sure if this one is an actual issue. Maybe you know?
6. Keyboard input is processed in the `Update` loop. Unsure if this could cause issues with inconsistent execution order. See previous point.

So those are generally my thoughts on it, and why it feels like a good change to me.

Reply:
> I tried to track when and where Draw is called, but couldn't get a clear answer. So I am not sure if this one is an actual issue. Maybe you know?

By default, every tick, the game does one update (`Update(UpdateTime);`), followed by one draw (`Draw(DrawTime);`):
https://github.com/stride3d/stride/blob/4c46fdcc4da0021514fbaaea3d432095d164b9b6/sources/engine/Stride.Games/GameBase.cs#L636-L663

`Draw` calls `GameSystems.Draw(gameTime);`:
https://github.com/stride3d/stride/blob/4c46fdcc4da0021514fbaaea3d432095d164b9b6/sources/engine/Stride.Games/GameBase.cs#L793-L807

> If the Draw and Update loops are not synced (always happening one right after the other), then pointer inputs could either be missed or processed multiple times

This would also occur within update if a user press and release a button faster than an update can be processed no down input would be registered.

The lowest draw rate for all games is 30 draws per second, anything below that is not comfortable.
By default, games run one update per draw, update first, then draw, if the developer decides to reduce the amount of updates per draw, there is no change in the logic if you where to run it inside update or draw, if the game is setup to update more often than draw `CompactPointerEvents` should collect all events that occured since the last draw into a single one, as you say, it might miss a pointer down, but that's only when they happen to last for less than 1/30th of a second if drawing at 30fps, which is *extremely* unlikely.

Reply:
This is something that can be improved then, you could append pointer events every update, and compact them on every draw right before usage.

Reply:
Perhapse, but then means we collect it all in `UISystem`, and then pass it to the `UIRenderFeature`, adding it as a dependency to `UISystem`. Though it is a workable soltuion, and should be done if we end up reverting the related changes in this PR I think.

With that said, I still feel like my other points about inconsistent execution order of input events, it causing some routed events to be raised in Draw and some in Update, and the lack of seperation of concern all still stand. And could cause unexpected issues that are hard to debug in the future.